### PR TITLE
Properly handle Run product merging in LHESource

### PR DESCRIPTION
**Pull request by Dr15Jones**
The LHESource was not properly handling the merging of the LHERunInfoProduct when multiple files are read. The framework requires that the source explicilty tell the framework a new file is being opened which setup the system for proper merging.

This change also fixed a crash in the LHESource which happened if multiple files were read.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

File: GeneratorInterface/LHEInterface/test/testMerge.log [hunk 1]
```diff
@@ -0,0 +1,153 @@
+Number of particles = 12
```
Comment (perrozzi) [inline GeneratorInterface/LHEInterface/test/testMerge.log]: hi, is this log file needed?

File: GeneratorInterface/LHEInterface/test/testNoMerge.log [hunk 2]
```diff
@@ -0,0 +1,197 @@
+Number of particles = 12
```
Comment (perrozzi) [inline GeneratorInterface/LHEInterface/test/testNoMerge.log]: same here

File: GeneratorInterface/LHEInterface/test/testMerge.log (hunk 1 above)
Comment (Dr15Jones) [reply GeneratorInterface/LHEInterface/test/testMerge.log]: It is used by the unit test. The test compares the output of the test job to this log file and if there is a difference, the test fails.

File: GeneratorInterface/LHEInterface/test/testNoMerge.log (hunk 2 above)
Comment (Dr15Jones) [reply GeneratorInterface/LHEInterface/test/testNoMerge.log]: This is also used by the unit test.

**Comment by perrozzi** [comment]
+1

**Comment by Dr15Jones** [comment]
Fixes issue #21807

**Comment by fabiocos** [comment]
please test workflow 513.0

**Comment by fabiocos** [comment]
+1

**Pull request merged**